massa-bootstrap/src/server.rs

```diff
@@ -355,23 +355,27 @@ pub async fn stream_bootstrap_information(
                 .get_executed_ops_part(last_ops_step);
             exec_ops_part = ops_data;
 
-            if let Some(slot) = last_slot && slot != final_state_read.slot {
-                if slot > final_state_read.slot {
-                    return Err(BootstrapError::GeneralError(
-                        "Bootstrap cursor set to future slot".to_string(),
-                    ));
+           final_state_changes = if let Some(slot) = last_slot  {
+                if slot != final_state_read.slot {
```

Comment:
Why not leaving this in the upper line with a `&&` and so having one `else` as before ?

Reply:
Intellij show error on this line : 
![image](https://user-images.githubusercontent.com/9607217/204281103-6372b77e-3f60-4aa7-88aa-671a62845d3d.png)

and 

![image](https://user-images.githubusercontent.com/9607217/204281243-1f97b293-5421-49ab-814a-6d6304be3705.png)

i think it just interpretor error.
I splited the pull request in two commit, i can drop the second or you can cherry pick the first. Like you want

Reply:
I think it's better is you drop the second one. Maybe your intellij doesn't support condition after if let.

Reply:
well it's done :)